docs/content/basic/authenticating-oauth.md

```diff
@@ -4,256 +4,703 @@ lang: en
+- `clientSecret`: secret value to confirm the client ID.
+- `stateSecret`: secret value used for [state verificiation][verification] of
+  authentication requests.
+- `scopes`: permissions requested for the `bot` user during installation.
+  [Explore scopes][scopes].
+- `installationStore`: handlers that store, fetch, and delete installation
+  information.
 
-We've provided a default implementation of the `installationStore`  `FileInstallationStore` which you can use during app development and testing.
+#### Development and testing
+
+Here we've provided a default implementation of the `installationStore` with
+[`FileInstallationStore`][store-file-installation] which can be useful when
+developing and testing your app:
 
 ```javascript
-const { App } = require('@slack/bolt');
-const { FileInstallationStore } = require('@slack/oauth');
+const { App } = require("@slack/bolt");
+const { FileInstallationStore } = require("@slack/oauth");
 const app = new App({
   signingSecret: process.env.SLACK_SIGNING_SECRET,
   clientId: process.env.SLACK_CLIENT_ID,
   clientSecret: process.env.SLACK_CLIENT_SECRET,
-  stateSecret: 'my-state-secret',
-  scopes: ['channels:history', 'chat:write', 'commands'],
+  stateSecret: "my-state-secret",
+  scopes: ["channels:history", "chat:write", "commands"],
   installationStore: new FileInstallationStore(),
 });
 ```
-:::warning 
 
-This is **_not_** recommended for use in production - you should implement your own production store. Please see the example code to the right and [our other examples](https://github.com/slackapi/bolt-js/tree/main/examples/oauth).
+:::warning
+
+This is **not** recommended for use in production - you should
+[implement your own installation store](#installation-store). Please continue
+reading or inspect [our example apps][examples].
 
 :::
 
+### Installer options
+
+We provide several options for customizing default OAuth using the
+`installerOptions` object, which can be passed in during the initialization of
+`App`. You can override these common options and
+[find others here][install-provider-options]:
+
+- `authVersion`: Settings for either new Slack Apps (`v2`) or Classic Slack Apps
+  (`v1`). Default: `v2`.
+- `directInstall`: Skip the [default installation page][installation] at
+  `installPath`. Default: `false`.
+- `installPath`: Path of the URL for starting an installation. Default:
+  `/slack/install`.
+- `metadata`: Relevant session information passed between requests. Optional.
+- `redirectUriPath`: Path of the installation callback URL. Default:
+  `/slack/oauth_redirect`.
+- `stateVerification`: Option to skip state verification for requests. Default:
+  `true`.
+- `userScopes`: User scopes to request during installation. Default: `[]`.
+- `callbackOptions`: Customized [responses to send][callbacks] during OAuth.
+  Default: [`CallbackOptions`][callback-options-default].
+- `stateStore`: Replace the `stateSecret` with a [custom state store][state].
```

Comment:
This is a module, which generates and verifies state parameter. Thus, I don't think this description explains what it is well.

We can say:
Customizable OAuth state parameter handling module; ClearStateStore should work for most scenarios. However, if you need even better security, implementing your own module that stores the state parameter data on the server-side database would be a good approach.

Reply:
Good to know we can recommend `ClearStateStore` with confidence! It reads much better when this is treated as a true module too, thanks for catching this!